cirq/protocols/json_serialization.py

```diff
@@ -44,175 +46,167 @@
-                'PasqalDevice': cirq.pasqal.PasqalDevice,
-                'PasqalVirtualDevice': cirq.pasqal.PasqalVirtualDevice,
-                'PauliString': cirq.PauliString,
-                'PhaseDampingChannel': cirq.PhaseDampingChannel,
-                'PhaseFlipChannel': cirq.PhaseFlipChannel,
-                'PhaseGradientGate': cirq.PhaseGradientGate,
-                'PhasedFSimGate': cirq.PhasedFSimGate,
-                'PhasedISwapPowGate': cirq.PhasedISwapPowGate,
-                'PhasedXPowGate': cirq.PhasedXPowGate,
-                'PhasedXZGate': cirq.PhasedXZGate,
-                'PhysicalZTag': cirq.google.PhysicalZTag,
-                'RandomGateChannel': cirq.RandomGateChannel,
-                'QuantumFourierTransformGate': cirq.QuantumFourierTransformGate,
-                'ResetChannel': cirq.ResetChannel,
-                'SingleQubitMatrixGate': single_qubit_matrix_gate,
-                'SingleQubitPauliStringGateOperation':
-                cirq.SingleQubitPauliStringGateOperation,
-                'SingleQubitReadoutCalibrationResult':
-                cirq.experiments.SingleQubitReadoutCalibrationResult,
-                'StabilizerStateChForm': cirq.StabilizerStateChForm,
-                'SwapPowGate': cirq.SwapPowGate,
-                'SycamoreGate': cirq.google.SycamoreGate,
-                'TaggedOperation': cirq.TaggedOperation,
-                'ThreeDQubit': cirq.pasqal.ThreeDQubit,
-                'Result': cirq.Result,
-                'TrialResult': cirq.TrialResult,
-                'TwoDQubit': cirq.pasqal.TwoDQubit,
-                'TwoQubitMatrixGate': two_qubit_matrix_gate,
-                'TwoQubitDiagonalGate': cirq.TwoQubitDiagonalGate,
-                '_UnconstrainedDevice':
-                cirq.devices.unconstrained_device._UnconstrainedDevice,
-                'VirtualTag': cirq.VirtualTag,
-                'WaitGate': cirq.WaitGate,
-                '_QubitAsQid': raw_types._QubitAsQid,
-                'XPowGate': cirq.XPowGate,
-                'XXPowGate': cirq.XXPowGate,
-                'YPowGate': cirq.YPowGate,
-                'YYPowGate': cirq.YYPowGate,
-                'ZPowGate': cirq.ZPowGate,
-                'ZZPowGate': cirq.ZZPowGate,
-
-                # not a cirq class, but treated as one:
-                'pandas.DataFrame': pd.DataFrame,
-                'pandas.Index': pd.Index,
-                'pandas.MultiIndex': pd.MultiIndex.from_tuples,
-                'sympy.Symbol': sympy.Symbol,
-                'sympy.Add': lambda args: sympy.Add(*args),
-                'sympy.Mul': lambda args: sympy.Mul(*args),
-                'sympy.Pow': lambda args: sympy.Pow(*args),
-                'sympy.Float': lambda approx: sympy.Float(approx),
-                'sympy.Integer': sympy.Integer,
-                'sympy.Rational': sympy.Rational,
-                'complex': complex,
-            }
-        return self._crd
-
-
-RESOLVER_CACHE = _ResolverCache()
```

Comment:
this is technically a backwards incompatible change. It may affect packages that layer on additional json-resolution paths on top of the base Cirq objects. However, I think those use cases *should* be solved by using `DEFAULT_RESOLVERS` below

Reply:
e.g. https://github.com/quantumlib/ReCirq/blob/master/recirq/serialization_utils.py#L225

Reply:
+1